clang/test/Sema/constant_builtins_vector.cpp

```diff
@@ -0,0 +1,99 @@
+// RUN: %clang_cc1 -verify -std=c++2a -fsyntax-only %s
+// expected-no-diagnostics
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
Can we clean up this test?  All of the macros doing the 'check' generation make this pretty unreadable.

Reply:
I just really didn't want to type out all the combinations. I can expand the macros out, np!

Reply:
I'm sorry @sethp, I don't follow your explanation, I'm not familiar enough with the testing infrastructure in Clang. If you think that will make the code look cleaner while still not having to spell out all convinations, I'm all for it!

Reply:
If you could provide a toy example, I can do that for this test

Reply:
My suggestion was not so much a way to avoid spelling them all out as to trick the preprocessor into doing (a lot of) the work. I'll try to put together a toy example in the next couple of days, but if you run `clang -E` on the existing file and commit the resulting spellings you'll get much of the way to what I was thinking. 

Would that address your concern, @erichkeane?

Reply:
Well, I got a quick demo of what I was thinking put together. For your consideration: https://github.com/llvm/llvm-project/commit/17c78a28b4f441bf284bcf6935061938ec908f31

It did & didn't work out quite the way I'd hoped: I'd forgotten just how, uh, _peculiar_ the preprocessor is, especially if you're trying to get it to generate human-readable output. There might be a way to do it nicely (adding `clang-format` into the pipeline?), or it might not be an idea worth pursuing at all. I'll leave the final determination up to you & @erichkeane .

Reply:
No, i don't think that is worth doing.  I'm still disturbed by how difficult it is to figure out exactly what/how we are testing things, but perhaps just some detailed comments on the macros are sufficient.